data/backend/authentication/views.py

```diff
@@ -0,0 +1,188 @@
+            )
+            if form.is_valid():
+                form.save()
+                return JsonResponse(
+                    {
+                        "success": True,
+                        "msg": _("The link is valid for %s min")
+                        % (getattr(settings, "LOGIN_URL_TIMEOUT", 900) / 60),
+                    }
+                )
+            else:
+                return JsonResponse({"success": False, "msg": _("Invalid form data")})
+        else:
+            form = self.get_form()
+            if form.is_valid():
+                return self.form_valid(form)
+            else:
+                return self.form_invalid(form)
+
+
+class RegistrationRequestedView(generic.TemplateView):
+    """confirmation that the email has beed stored"""
+
+    template_name = "registration/registration_requested.html"
+
+
+class Activate(MailauthLoginTokenView):
+    """login and activate the disabled account"""
+
+    signer = signing.UserSigner()
+    max_age = getattr(settings, "LOGIN_URL_TIMEOUT", 60 * 15)
+    single_use = getattr(settings, "LOGIN_TOKEN_SINGLE_USE", True)
+    success_url = getattr(settings, "LOGIN_REDIRECT_URL", "authentication:profile")
+
+    def get_success_url(self):
+        return "/"
+        # TODO redirect to profile instead, to give the occasion to fill in more details:
+        # return resolve_url(self.success_url)
+
+    def get(self, request, *__, **kwargs):
+        token = kwargs["token"]
+
+        try:
+            user = self.signer.unsign(
+                token, max_age=self.max_age, single_use=self.single_use
+            )
+            user.is_active = True
+            user.save()
+        except:
+            msg = _("The token has expired")
+            logger.warning(msg, exc_info=True)
+            raise PermissionDenied
+
+        user = authenticate(request, token=token)
+        if user is None:
+            raise PermissionDenied
+        else:
+            login(self.request, user=user)
+            # Remove token from the HTTP Referer header
+            self.request.path.replace(token, "login-token")
+
+        return response.HttpResponseRedirect(self.get_success_url())
+
+
+def profile(request):
+    if request.method == "GET":
+        u = request.user
+        if u.is_authenticated:
+            return JsonResponse(
+                {
+                    "email": u.email,
+                    "first_name": u.first_name,
+                    "last_name": u.last_name,
+                    "organization": u.organization,
+                    "terms_of_use": u.terms_of_use,
+                    "token": u.token,
+                }
+            )
+        else:
+            return JsonResponse(
+                {"error": _("You must be authenticated to access this page")},
+                status=401,
+            )
+
+
+def is_token_valid(token):
+    return EcobalyseUser.objects.filter(token=token)
```

Comment:
We should return a bool here, not an expensive collection; eg. `return EcobalyseUser.objects.filter(token=token).count() > 0`